### DISCUSSION TITLE
How to make Accordion Trigger Sticky

### DISCUSSION THREAD

**Discussion by victormongi** [shadcn-ui/ui#4206] [General]
<img width="660" alt="Screenshot 2024-07-07 at 9 01 27 PM" src="https://github.com/shadcn-ui/ui/assets/5617746/c637d256-3eca-48c2-90c5-7052ecf62a9f">

is there any way to make this sticky?

Thank you all.

**Comment by viniciusofp** [comment]
Just had to face this, here is what I've found:

When the element is rendered, AccordionTrigger is wrapped inside a h3. Like this:

```html
<h3
  data-orientation="vertical"
  data-state="open"
  class="flex"
>
  // This button is AccordionTrigger
  <button
    type="button"
    aria-controls="radix-:r14:"
    aria-expanded="true"
    data-state="open"
    data-orientation="vertical"
    id="radix-:r13:"
    class="..." // Applying "sticky top-0" here won't work as expected because h3 is the parent, not AccordionItem.
    data-radix-collection-item=""
  >
    ...
  </button>
</h3>
```

I was able to make it work by wrapping AccordionTrigger inside a div with "sticky top-?" classes. Here is the code in my case:

```tsx
<div className="sticky top-0">
  <AccordionTrigger>
    ...
  </AccordionTrigger>
</div>
```

**Reply by victormongi** [reply]
Thank you ill check that out
